demo/src/app/shared/code/code.component.ts

```diff
@@ -6,19 +6,36 @@ import {CodeHighlightService} from './code-highlight.service';
   selector: 'ngbd-code',
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <pre class="language-{{ lang }}"><code #code class="language-{{ lang }}"></code></pre>
+    <pre
```

Comment:
please rollback to the version without `[ngClass]`, this works perfectly fine:

```html
<pre class="language-{{ lang }}"><code #codeEl class="language-{{ lang }}">{{code.trim()}}</code></pre>
```

Reply:
Well, I think it is easier to use `ngClass` when the class is complex, and it when adding line numbering. Actually I started to remove line numbering too following your comments, but I did half of the job. However I maintain that I'd like to keep it if the numbering doesn't bring performance impact (also I don't know which order of magnitude we are talking about).

Reply:
here we have a single class, binding was just fine

Reply:
We won't have a single class anymore if keeping the **line highlighting**, and this is a pending item, therefore I won't go back and forth in changing the code. We are not talking about a final code here as long as this item is pending.